rollouts/controllers/rollout_controller.go

```diff
@@ -192,43 +192,56 @@ func (r *RolloutReconciler) validateProgressiveRolloutStrategy(ctx context.Conte
 		return err
 	}
 
-	clusterWaveMap := make(map[string]int)
+	clusterWaveMap := make(map[string]string)
 	for _, cluster := range allClusters.Items {
-		clusterWaveMap[cluster.Name] = -1
+		clusterWaveMap[cluster.Name] = ""
 	}
 
-	for waveIdx, wave := range strategy.Spec.Waves {
+	pauseAfterWaveName := ""
+	pauseWaveNameFound := false
+
+	if rollout.Spec.Strategy.Progressive != nil {
+		pauseAfterWaveName = rollout.Spec.Strategy.Progressive.PauseAfterWave.WaveName
+	}
+
+	for _, wave := range strategy.Spec.Waves {
 		waveClusters, err := r.store.ListClusters(ctx, rollout.Spec.Targets.Selector, wave.Targets.Selector)
 		if err != nil {
 			return err
 		}
 
 		if len(waveClusters.Items) == 0 {
-			return fmt.Errorf("wave %d does not target any clusters", waveIdx)
+			return fmt.Errorf("wave '%s' does not target any clusters", wave.Name)
 		}
 
 		for _, cluster := range waveClusters.Items {
 			currentClusterWave, found := clusterWaveMap[cluster.Name]
 			if !found {
 				// this should never happen
-				return fmt.Errorf("wave %d references cluster %s not selected by the rollout", waveIdx, cluster.Name)
+				return fmt.Errorf("wave '%s' references cluster %s not selected by the rollout", wave.Name, cluster.Name)
```

Comment:
tip: %q will give you quoted value of the arg.

Reply:
Thanks! This is really good to know. I updated the print statements to use %q instead of '%s' for rollout name.